### DISCUSSION TITLE
Why are devices (Wyze Lock) supported by ZHA but not z2m and seem to require reverse engineering?

### DISCUSSION THREAD

**Discussion by eigenphase** [Koenkk/zigbee2mqtt#14090] [Devices]
I have a WyzeLock which worked like a charm with Home Assistant and ZHA (https://zigbee.blakadder.com/Wyze_Lock.html)

I recently migrated to z2m whose architecture I like far better than ZHA. All devices work well EXCEPT WyzeLock. It pairs but only has issues (sometimes I get spurious locks/unlocks, the lock state is not reported etc).

In a github issue https://github.com/Koenkk/zigbee2mqtt/issues/3018 , raw bytes from the WyzeLock were reverse engineered and a filter was created for z2m (which is not as reliable as ZHA, see above).

So my question is: Why is it necessary to start from scratch and do reverse engineering when a device is fully supported by ZHA, another open source project? Couldn't we just take the info needed from there?

(PS: This is really not a blame for z2m, I am genuinely curious why this is the case)

**Comment by DubhAd** [comment]
Nothing is stopping _you_ from doing that - at least assuming that the ZHA quirks are licensed in a way that allows that.

However, it's not possible to use the ZHA quirk directly, because they're two totally different pieces of software, with different expectations and requirements. ZHA defaults to assuming that devices will follow the Zigbee spec and uses quirks when that doesn't happen (which is typically the case). Z2M assumes nothing will work without help, and so requires a converter for everything.
